tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3134,4 +3134,52 @@ public function publicDownloadsTheFolderFromTheLastCreatedPublicLink(string $res
 			)
 		);
 	}
+
+	/**
+	 * @Then /^the json responded should contain these key and value pairs:$/
+	 *
+	 * Loops through data items in response to find given key and value pairs
+	 * It does not support field value replacing
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theJsonRespondedShouldContainTheseKeyAndValuePairs(TableNode $table):void {
+		$this->featureContext->verifyTableNodeColumns($table, ['key', 'value']);
+		$response = json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
+		$data = $response["ocs"]["data"];
+		foreach ($table->getHash() as $row) {
+			$key = $row['key'];
+			$value = $row['value'];
+			$segments = explode('@@@', $key);
+			if (\sizeof($segments) > 1) {
+				$arrayKeyExists = \array_key_exists($segments[0], $data);
+				Assert::assertTrue($arrayKeyExists, "The key $key does not exist on the response");
+				$final_key_value  = $data[$segments[0]];
+				$last_array =  $data[$segments[0]];
+				// Move inside nested blocks each step at a time
+				for ($i = 1;$i < \sizeof($segments);$i++) {
```

Comment:
```suggestion
				for ($i = 1;$i < \sizeof($segments);$i++) {
```
for variable `$i` give some suitable name